File: 240313/행복한 수열의 개수/number-of-happy-sequence.cpp
#include <iostream>
#include <algorithm>
using namespace std;
#define MAX_NM 100

int n, m;
int arr[MAX_NM][MAX_NM];

int main() {
    // 여기에 코드를 작성해주세요.
    cin >> n>>m;
    for(int i=0; i<n; i++) {
        for(int j=0; j<n; j++) {
            cin >> arr[i][j];
        }
    }

    // if(n==1 && m==1) {
    //     cout << 2;
    //     return 0;
    // }

    int ans = 0;
    // 각 행별로 확인하기
    for(int i=0; i<n; i++) {
        int cnt = 1, max_seq = 1;
        for(int j=1; j<n; j++) {
            if(arr[i][j-1] == arr[i][j]) cnt++;
            else cnt = 1;

            // n이 1이고 m도 1일때의 대비
            max_seq = max(max_seq, cnt);
        }

        if(max_seq >= m) ans++;
    }

    // cout << ans << '\n';

    // 각 열별로 확인하기
    for(int i=0; i<n; i++) {
        int cnt = 1, max_seq = 1;
        for(int j=1; j<n; j++) {
            if(arr[j-1][i] == arr[j][i]) cnt++;
            else cnt = 1;

            max_seq = max(max_seq, cnt);
        }
        if(max_seq >= m) ans++;
    }

    cout << ans;
    return 0;
}

File: 241104/행복한 수열의 개수/number-of-happy-sequence.py
n, m = tuple(map(int, input().split()))

grid = [
    list(map(int, input().split()))
    for _ in range(n)
]

arr = [0 for _ in range(n)]

ans = 0

def checkArr():
    max_cnt = 1
    cnt = 1
    for i in range(1, n):
        if arr[i] == arr[i-1]:
            cnt += 1
        else:
            cnt = 1

        max_cnt = max(max_cnt, cnt)
    
    return max_cnt >= m

for i in range(n):
    arr = grid[i][:]
    if checkArr():
        ans += 1
    
    for j in range(n):
        arr[j] = grid[j][i]
    
    if checkArr():
        ans += 1

print(ans)